### DISCUSSION TITLE
Issue with Azure DNS Zone Permissions?

### DISCUSSION THREAD

**Discussion by RiaanStrydom198** [polymind-inc/acmebot#333] [Q&A] [answered]
Hi there.  Hopefully someone here can assist.
I have deployed the App as Documented, and followed everything exactly as described.  I currently host my DNS with Azure DNS, and have also given the Function App the required DNS Zone Contributor Permissions on the DNS Zone.
However, when I go to the /add-certificate URL to issue the cert, it starts loading the DNS Zone and then after a few seconds throws the attached error:

![AcmeFunctionAppError](https://user-images.githubusercontent.com/78362787/122426101-441e2a80-cf90-11eb-8c0b-de3729be9f72.png)

Is the error related to DNS Permissions, or am I missing something else?

Thanks in advance!

**Comment by shibayan** [comment, marked as answer]
I would like to fix the problem of the error messages not being displayed correctly.

The error log is output to Application Insights, which was created at the same time as Acmebot, so could you please check that?

**Reply by RiaanStrydom198** [reply]
Great - thanks a mil! I can see this error in the App Insights log:

![image](https://user-images.githubusercontent.com/78362787/122511395-74ed7680-d007-11eb-95a5-9035d4edd826.png)

It seems to definitely refer to DNS Zone access right?  That's strange as I gave the Azure Function contributor as well as DNS Zone Contributor access to the DNS Zone in question as well as the Resource group in which the DNS Zone (As well as the function) resides.

Any ideas?

**Reply by RiaanStrydom198** [reply]
There actually seems to be two errors thrown each time I try and load the page:

![image](https://user-images.githubusercontent.com/78362787/122512380-0d382b00-d009-11eb-9672-e7ef972b9b27.png)

**Reply by shibayan** [reply]
Are you seeing Exception or Dependency logs in Application Insights?

**Reply by RiaanStrydom198** [reply]
Does this help? Happy to export all the logs if you could possibly assist with the proper query to give the info you need?

![image](https://user-images.githubusercontent.com/78362787/122513732-15916580-d00b-11eb-962c-d7e4eb469029.png)

**Reply by RiaanStrydom198** [reply]
Just found this exception too:

![image](https://user-images.githubusercontent.com/78362787/122513935-5d17f180-d00b-11eb-943a-298e0147b5ab.png)

What is meant by "DNS Provider is not configured"? I did add the correct permissions to the DNS Zone - Is there another configuration i am perhaps missing?

**Reply by shibayan** [reply]
Oh, I understand. Please refer to this page to configure your DNS provider, and for Azure DNS, you need to configure your subscription ID.
https://github.com/shibayan/keyvault-acmebot/wiki/DNS-Provider-Configuration#azure-dns

**Reply by RiaanStrydom198** [reply]
Fantastic, thanks! I added the following configuration to the function and it worked immediately:

![image](https://user-images.githubusercontent.com/78362787/122515155-4ecad500-d00d-11eb-8577-68df85767550.png)

Just a suggestion - The documentation could perhaps be updated to be a little clearer around this configuration needed for Azure DNS?  I am happy to contribute and do a quick write-up with screenshots if that will help you? Also keen to help wherever I can, this is such a great project!

**Reply by shibayan** [reply]
It is mentioned in the README, but it is small and may be easily overlooked. Your help in improving the documentation is always welcome.
https://github.com/shibayan/keyvault-acmebot#3-add-settings-for-your-choice-dns-provider

**Reply by robincher** [reply]
Got the same issue, and thank god i found this out.  Will be happy to create a PR to further improve the documents :)
